### ISSUE TITLE
`getNotesForVoting` performs invalid `AddressIndex` comparison

### ISSUE THREAD

**Issue by turbocrime** [penumbra-zone/web#1638]
https://github.com/penumbra-zone/web/blob/69b083755e83a5d9dc5465345df7981293fb0b1b/packages/storage/src/indexed-db/index.ts#L460-L469

similar to https://github.com/penumbra-zone/web/issues/1604 and https://github.com/penumbra-zone/web/issues/1601

**Comment by grod220** [comment]
Yes, we should create a hasTokenBalance helper here and write a test case for this. The reason why no users would realistically run into this is because it would only error if someone ibc'd out a delegation token and ibc'd it back in to an ephemeral address.

**Comment by Valentine1898** [comment]
> The reason why no users would realistically run into this is because it would only error if someone ibc'd out a delegation token and ibc'd it back in to an ephemeral address.

delUM can simply be sent to the ephemeral address, it is not necessary to use ibc out + ibc in to do this
This issue is not a priority anyway, because the issue is only `notesForVoting` RPC (which probably nobody uses), while planner ignores `randomizer` 
https://github.com/penumbra-zone/web/blob/2c0c167430fb39643761ecb812e80fe9942e7e10/packages/wasm/crate/src/storage.rs#L103-L108
